src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -84,21 +93,38 @@ public function __construct( $id_base = '', $name = '', $widget_options = [], $c
 			Arr::get( $arguments, 'control_options', [] )
 		);
 
-		// @todo add what this does in in TEC-3612 & TEC-3613.
 		$this->setup();
 	}
 
 	/**
 	 * Setup the widget.
 	 *
-	 * @todo update in TEC-3612 & TEC-3613
-	 *
 	 * @since TBD
 	 *
 	 * @return mixed
 	 */
 	public abstract function setup();
 
+	/**
+	 * {@inheritDoc}
+	 */
+	public function form( $instance ) {
+
+		add_filter(
+			"tribe_widget_{$this->get_registration_slug()}_arguments",
+			static function ( array $arguments ) use ( $instance ) {
```

Comment:
This has been refactored. Although the closure was used here because it should be near impossible to unhook as this was setting up using the saved data for the widget. So unhooking it, is not something we should support. 

There is still refactoring to do, with this method to make the parent and child method work together.

Reply:
Unhooking is not recommended, and we do not need to support it, but it should be possible.
Before PHP 5.6, and Closures, we had plenty of code that should have not been remover or unhooked that we could still remove.
Furthermore, none of us know what we might need this for in the future.